ci_framework/roles/openshift_provisioner_node/tasks/add_user.yml

```diff
@@ -0,0 +1,59 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: "Adding {{ cifmw_opn_user }} to the system"
+  become: true
+  ansible.builtin.user:
+    name: "{{ cifmw_opn_user }}"
+    comment: "OpenShift installer"
+    append: true
+    groups: libvirt
+    state: present
+
+- name: Generate SSH keys
+  delegate_to: localhost
+  block:
+    - name: "Generate SSH RSA Key for {{ cifmw_opn_user }}"
+      when:
+        - cifmw_use_hive is defined
+        - cifmw_use_hive | bool
+      register: cifmw_opn_user_rsa_key
+      community.crypto.openssh_keypair:
+        path: "{{ cifmw_opn_artifacts_dir }}/{{ cifmw_opn_user }}_id_rsa"
+        size: 2048
+
+    - name: "Generate SSH ED25519 Key for {{ cifmw_opn_user }}"
+      register: cifmw_opn_user_ed25519_key
+      community.crypto.openssh_keypair:
+        path: "{{ cifmw_opn_artifacts_dir }}/{{ cifmw_opn_user }}_id_ed25519"
+        type: "ed25519"
```

Comment:
why generate both types? We can probably only rely on the ed25519.. ?

Reply:
It is a limitation in Hive operator. For doing libvirt remote management, it accepts only SSH keys generated with RSA algorithm. 

For accessing the host in RHEL 9.2 , we need SSH keys other than RSA. Hence the two types.